#! /bin/sh /usr/share/dpatch/dpatch-run
## 102_gtk_zero_pagesize.dpatch by Artem Popov <artfwo@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Cherry picks from 0.47 SVN for the color chooser (LP: #257685)
## DP: and GtkAdjustment related warnings.

@DPATCH@
diff -urNad inkscape-0.46~/src/dialogs/export.cpp inkscape-0.46/src/dialogs/export.cpp
--- inkscape-0.46~/src/dialogs/export.cpp	2008-03-11 10:19:54.000000000 +0600
+++ inkscape-0.46/src/dialogs/export.cpp	2009-04-08 19:51:19.128959416 +0700
@@ -207,7 +207,7 @@
                            int digits, unsigned int sensitive,
                            GCallback cb, GtkWidget *dlg )
 {
-    GtkObject *a = gtk_adjustment_new (val, min, max, step, page, page);
+    GtkObject *a = gtk_adjustment_new (val, min, max, step, page, 0);
     gtk_object_set_data (a, "key", key);
     gtk_object_set_data (GTK_OBJECT (dlg), (const gchar *)key, a);
 
diff -urNad inkscape-0.46~/src/widgets/select-toolbar.cpp inkscape-0.46/src/widgets/select-toolbar.cpp
--- inkscape-0.46~/src/widgets/select-toolbar.cpp	2008-03-11 10:20:17.000000000 +0600
+++ inkscape-0.46/src/widgets/select-toolbar.cpp	2009-04-08 19:53:25.496128202 +0700
@@ -255,7 +255,7 @@
                                                        gchar const *tooltip,
                                                        gboolean altx )
 {
-    GtkAdjustment* adj = GTK_ADJUSTMENT( gtk_adjustment_new( 0.0, lower, 1e6, SPIN_STEP, SPIN_PAGE_STEP, SPIN_PAGE_STEP ) );
+    GtkAdjustment* adj = GTK_ADJUSTMENT( gtk_adjustment_new( 0.0, lower, 1e6, SPIN_STEP, SPIN_PAGE_STEP, 0 ) );
     if (tracker) {
         tracker->addAdjustment(adj);
     }
diff -urNad inkscape-0.46~/src/widgets/sp-color-slider.cpp inkscape-0.46/src/widgets/sp-color-slider.cpp
--- inkscape-0.46~/src/widgets/sp-color-slider.cpp	2009-04-08 19:50:22.884123859 +0700
+++ inkscape-0.46/src/widgets/sp-color-slider.cpp	2009-04-08 19:54:27.485123949 +0700
@@ -330,8 +330,14 @@
 	g_return_if_fail (SP_IS_COLOR_SLIDER (slider));
 
 	if (!adjustment) {
-		adjustment = (GtkAdjustment *) gtk_adjustment_new (0.0, 0.0, 1.0, 0.01, 0.1, 0.1);
+		adjustment = (GtkAdjustment *) gtk_adjustment_new (0.0, 0.0, 1.0, 0.01, 0.0, 0.0);
 	}
+#if GTK_CHECK_VERSION (2,14,0)
+    else {
+        gtk_adjustment_set_page_increment(adjustment, 0.0);
+        gtk_adjustment_set_page_size(adjustment, 0.0);
+    }
+#endif
 
 	if (slider->adjustment != adjustment) {
 		if (slider->adjustment) {
diff -urNad inkscape-0.46~/src/widgets/toolbox.cpp inkscape-0.46/src/widgets/toolbox.cpp
--- inkscape-0.46~/src/widgets/toolbox.cpp	2009-04-08 19:50:22.656133932 +0700
+++ inkscape-0.46/src/widgets/toolbox.cpp	2009-04-08 19:52:38.453121340 +0700
@@ -703,7 +703,7 @@
                                                        gdouble climb = 0.1, guint digits = 3, double factor = 1.0 )
 {
     GtkAdjustment* adj = GTK_ADJUSTMENT( gtk_adjustment_new( prefs_get_double_attribute(path, data, def) * factor,
-                                                             lower, upper, step, page, page ) );
+                                                             lower, upper, step, page, 0 ) );
     if (us) {
         sp_unit_selector_add_adjustment( SP_UNIT_SELECTOR(us), adj );
     }
